certbot-ci/certbot_integration_tests/certbot_tests/context.py

```diff
@@ -1,16 +1,134 @@
+"""Module do handle the context of integration tests."""
+import os
+import tempfile
+import subprocess
+import shutil
+import sys
+from distutils.version import LooseVersion
+
+from certbot_integration_tests.utils import misc
+
+
 class IntegrationTestsContext(object):
     """General fixture describing a certbot integration tests context"""
     def __init__(self, request):
         self.request = request
+
+        self.worker_id = request.config.slaveinput['slaveid'] if hasattr(request.config, 'slaveinput') else 'master'
         if hasattr(request.config, 'slaveinput'):  # Worker node
             self.worker_id = request.config.slaveinput['slaveid']
             self.acme_xdist = request.config.slaveinput['acme_xdist']
         else:  # Primary node
             self.worker_id = 'primary'
             self.acme_xdist = request.config.acme_xdist
+
+        self.acme_server =self.acme_xdist['acme_server']
         self.directory_url = self.acme_xdist['directory_url']
         self.tls_alpn_01_port = self.acme_xdist['https_port'][self.worker_id]
         self.http_01_port = self.acme_xdist['http_port'][self.worker_id]
+        self.challtestsrv_mgt_port = self.acme_xdist['challtestsrv_port']
+
+        self.certbot_version = misc.get_certbot_version()
+
+        self.workspace = tempfile.mkdtemp()
+        self.config_dir = os.path.join(self.workspace, 'conf')
+        self.hook_probe = tempfile.mkstemp(dir=self.workspace)[1]
+
+        self.manual_dns_auth_hook = (
```

Comment:
They will be called in several places in `test_main`. Their exact format depends on `challtestsrv_mgt_port`, that is part of the integration test context, so theses strings needs to be set up in the integration test context fixture.

I could put them as output of two methods in this fixture, but it seems to me to add useless complexity, since their value will not be change for a given instance of the fixture.

Reply:
Ok, makes sense.